doc/ext-pake/proposed/spake2plus_proposal_draft.md

```diff
@@ -0,0 +1,67 @@
+Introduction
+------------
+
+This document proposes protocol flow and high-level API changes required in PSA Cryptography API 1.1 PAKE Extension, to support the SPAKE2+ protocol.
+
+SPAKE2+ Version
+---------------
+
+SPAKE2+, an Augmented PAKE Draft 02, 10 December 2020 is considered for proposal.
+
+Link : [https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf](https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf)
+
+Remarks
+
+*   SPAKE2+, an Augmented PAKE Draft 08, 5 May 2022 is the latest draft version. Link : [https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08](https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08)
+*   Shared Secret Key generation is not compatible between Draft 02 and 08.
+*   As most SPAKE2+ implementations e.g. Matter Specification Version 1.0 are based on Draft 02, this version is being considered for better interoperability.
+
+Expected PAKE API Flow :
+------------------------------------
+![](SPAKE2PLUS.svg)
```

Comment:
Handling Y (= shareV) and cB (= confirmP) in two separate input/output passes with different step types (`PSA_PAKE_STEP_KEY_SHARE` and `PSA_PAKE_STEP_CONFIRM`) would be cleaner and simplify usage in cases where the two are transferred in separate protocol elements. Packing them together in a single package is still straight forward by calling both inputs or outputs immediately after each other.

Reply:
> Handling Y (= shareV) and cB (= confirmP) in two separate input/output passes with different step types (`PSA_PAKE_STEP_KEY_SHARE` and `PSA_PAKE_STEP_CONFIRM`) would be cleaner and simplify usage in cases where the two are transferred in separate protocol elements. Packing them together in a single package is still straight forward by calling both inputs or outputs immediately after each other.

Agreed. I think this diagram copies the style of the [J-PAKE sequence diagram](https://arm-software.github.io/psa-api/crypto/1.1/ext-pake/api/pake.html#c.PSA_ALG_JPAKE) - which does not explicitly show every call to `psa_pake_input()` or `psa_pake_output()` when multiple parameters are input or output from the operation.

But the associated text should specify that these values are input and output individually, and describe any ordering requirements.

Reply:
I will update the flow description to include this

Reply:
I have added an example Matter implementation which demonstrates the flow of the PAKE API.